### ISSUE TITLE
SteamVR Settings PSMove Controller buttons Override Problem

### ISSUE THREAD

**Issue by Cieles** [cboulay/PSMoveService#513]
Hey guys, 
I have the ps move set up to play steam vr games, I've modified the steamvr.vrsettings file to map out the steam vr controller.

> {
>   "driver_vridge": {
>     "fakeSensorEnabled": false,
>     "renderScale": 100,
>     "windowHeight": 1440,
>     "windowWidth": 2560
>   },
>   "psmove": {
>     "meters_per_touchpad_units": 0.075000002980232239,
>     "move": "touchpad",
>     "ps": "system",
>     "rumble_suppressed": true,
>     "select": "button_9",
>     "start": "button_10",
>     "trigger": "trigger",
>     "use_spatial_offset_after_touchpad_press_as_touchpad_axis": true
>   },
>   "psmove_touchpad_directions": {
>     "circle": "touchpad_right",
>     "cross": "touchpad_left",
>     "triangle": "touchpad_up",
>     "square": "touchpad_down"
>   },
>   "psmove_settings": {
>     "psmove_filter_hmd_serial": "04:76:6e:c0:8b:d6"
>   },
>   "steamvr": {
>     "activateMultipleDrivers": true,
>     "defaultMirrorView": 1,
>     "mirrorViewGeometry": "1249 473 1240 1165",
>     "renderTargetMultiplier": 1.7000000476837158
>   }
> }
> 

The only problem is when I am in game these controls don't actually do anything, I am not able to move. The game assumes I'm using the oculus touch when this type of mapping is aimed to be for the htc vive, I'm not sure if that is necessarily the focus of the problem. 
I noticed the keyboard keys 'wasd' works for moving about, so I went to look for ways to map the keyboard onto the psmove controllers. I found this program called joytokey in hopes to map keyboard keys to the psmove controller, the program detects the controllers but is not able to detect any of the keys I'm pressing on the ps move controller.

I've been stuck on this for days now, and I'm not sure what to do, I'm running out of options, please help!!

So to summarize...
1. Is there a way to map the keyboard keys onto the ps move controllers manually 
2. Is there a way to map oculus touch controls over the ps move controllers

Thanks in advance!

**Comment by pmantr** [comment]
Hello there,
from what I understand, psmove can be mapped to vive controllers only. If you haven't allready done this, take a look [here](https://github.com/HipsterSloth/PSMoveSteamVRBridge/wiki/Steam-VR-Setup).
For other mappings you might want to check out FreePIE, where you might be able to have more agility on mapping different HIDs, but I have never tried it myself.

On your second question, the answer is yes, and you should add this code into your steamvr.vrsettings 
`"psmove" : {
      "move" : "touchpad",
      "meters_per_touchpad_units" : 0.075,
      "use_spatial_offset_after_touchpad_press_as_touchpad_axis":true
   },`

If you follow this solution, when you hold down on move button, and move your hand around, directional touch will be emulated.

Next problem that you will find is mapping different functions to the two psmove controllers. eg X on right controller mapped to something different from X on the left controller.

Hope that this info helps, 
Keep Hacking

**Comment by cboulay** [comment]
I've never actually used PSMoveService with Steam. Please ask here: https://github.com/HipsterSloth/PSMoveSteamVRBridge

**Comment by meppers** [comment]
did you set the steamvr.vrsettings file to read only after you modified it?
(you need to)

**Comment by peterjordan1977** [comment]
Maybe i should have asked earlier as i have been stuck on this step for a while!
